packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -0,0 +1,319 @@
+        await Promise.all(this.contributions.getContributions()
+            .map(contribution => contribution.onQuit && contribution.onQuit(event, exitCode)));
+    }
+
+    protected bindApplicationEvents(app: electron.App): MaybePromise<void> {
+        // tslint:disable-next-line:no-any
+        app.on('activate', (event, hasVisibleWindows) => this.activate(event, hasVisibleWindows));
+        app.on('window-all-closed', () => this.windowAllClosed(app));
+        app.on('before-quit', event => this.beforeQuit(event));
+        app.on('quit', (event, exitCode) => this.quit(event, exitCode));
+    }
+
+    protected bindIpcEvents(): void {
+        electron.ipcMain.on('create-new-window',
+            (event: electron.IpcMessageEvent, url?: string) => this.createNewWindow(url));
+        electron.ipcMain.on('open-external',
+            (event: electron.IpcMessageEvent, url?: string) => url && this.openExternally(url));
+    }
+
+    protected activate(event: electron.Event, hasVisibleWindows: boolean): MaybePromise<void> {
+        if (electron.BrowserWindow.getAllWindows().length === 0) {
+            this.openWindow();
+        }
+    }
+
+    protected async startBackend(app: electron.App): Promise<void> {
+        const devMode = process.defaultApp || /node_modules[\/]electron[\/]/.test(process.execPath);
+        try {
+            await (devMode
+                ? this.requireBackend(app)
+                : this.forkBackend(app));
+        } catch (error) {
+            console.error(error);
+            app.exit(1);
+        }
+    }
+
+    /**
+     * In development mode, it is easier to run the backend as part of the main process.
+     *
+     * @param app
+     */
+    protected async requireBackend(app: electron.App): Promise<number> {
+        // tslint:disable-next-line:no-any
+        require(this.mainPath).then((address: any) => {
+            this.backendPortDeferred.resolve(address.port);
+        }).catch((error: Error) => {
+            this.backendPortDeferred.reject(error);
+        });
+        return this.backendPort;
+    }
+
+    /**
+     * In production mode, the backend gets its own process.
+     *
+     * @param app
+     */
+    protected async forkBackend(app: electron.App): Promise<number> {
+        const backendProcess = fork(this.mainPath);
+        backendProcess.on('error', error => {
+            this.backendPortDeferred.reject(error);
+        });
+        backendProcess.on('message', message => {
+            const port = Number.parseInt(message, 10);
+            if (!isNaN(port)) {
+                this.backendPortDeferred.resolve(port);
+            }
+        });
+        app.on('quit', () => {
+            process.kill(backendProcess.pid);
+        });
+        return this.backendPort;
+    }
+
+    protected windowAllClosed(app: electron.App): MaybePromise<void> {
+        // Usually, Mac application don't stop when there isn't more windows.
+        if (process.platform !== 'darwin') {
```

Comment:
Ah, this is because under Mac, applications seems to stay in stand by when all the windows are closed. There is an [event](https://electronjs.org/docs/api/app#event-activate-macos) one should listen to when the user requires a new window. But this is not implemented here indeed.

Do you prefer the backend to be shutdown completely on Mac while we don't listen for said event?

Reply:
I commented out the Mac specific escape, it should stop the backend.